coalib/output/TomlConfWriter.py

```diff
@@ -0,0 +1,177 @@
+import os
+
+from coala_utils.string_processing import unescape
+from coalib.parsing.TomlConfParser import TomlSetting
+
+from tomlkit import document, table, dumps, array, string, key, integer
+from tomlkit.items import (Array, String, Bool, Integer, Comment, Key,
+                           KeyType, Trivia)
+
+
+class TomlConfWriter:
+
+    """
+    Class that contains functions for
+    converting toml sections to toml
+    file
+    """
+
+    def __init__(self, file):
+        self.file = file
+        if os.path.isdir(self.file):
+            self.file = os.path.join(self.file, '.coafile.toml')
+        self.unsavable_keys = ['save', 'toml_config']
+        self.document = document()
+
+    def write(self, sections):
+        """
+        Converts the given sections into a TOML
+        document and writes the document into
+        the specified file
+
+        :param sections: The sections that have to be
+                         written into the file
+        """
+
+        if sections.get('cli', None):
+            self.remove_unsavable_settings(sections)
+
+        for item in sections:
+            section = sections[item]
+            table_name = self.get_table_name(section)
+            if table_name in self.document:
+                continue
+            table_contents = table()
+            for _, setting in section.contents.items():
+
+                setting_key = self.get_setting_key(setting)
+
+                if isinstance(setting, TomlSetting):
+                    value = setting.original_value
+                else:
+                    value = self.get_original_value(setting.value)
```

Comment:
`setting.original_value` can work if the setting is a TomlSetting (obtained from a toml file).
But  `coala -s` will invoke the `TomlConfWriter`  by which settings from the command line will be saved as well into the toml file as well. Since these are not  `TomlSetting`, they do not have the `original_value` parameter and data-type has to be guessed.